libs/crosstarget-utils/src/wasm/spawn.rs

```diff
@@ -1,10 +1,12 @@
 use std::future::Future;
 
+use futures::FutureExt;
+
 use crate::common::SpawnError;
 
-pub async fn spawn_if_possible<F>(future: F) -> Result<F::Output, SpawnError>
+pub fn spawn_if_possible<F>(future: F) -> impl Future<Output = Result<F::Output, SpawnError>>
 where
     F: Future + 'static,
 {
-    Ok(future.await)
+    future.map(Ok)
```

Comment:
Unrelated to this PR but is there a reason this is not using `wasm_bindgen_futures::spawn_local`?

Reply:
Mostly, in the places `spawn_if_possible` is used it won't provide an actual benefit over just awaiting the future - ideally we want to do multiple tasks in parallel there, not just async.

Reply:
My concern is that it makes the semantics of `spawn_if_possible` different depending on if it's compiled for native code or for WASM: the native version will always execute the future regardless of whether the result future is ever polled, the WASM version won't. The same goes for dropping the future: the two versions have different cancellation safety properties which is important, e.g., when using it in the `select!` macro. Even if this is not a problem in current code, it has potential for bugs in the future.

Again, this is not blocking this PR as it's out of scope here. But I think the behavior should be unified if possible.

Reply:
Feel free to ignore this if this difference is intended and is implied by `if_possible` in the name, but I feel a doc comment explaining this would be helpful in that case.

Reply:
That's a great point, I think we are not in aa rush to get the fix ASAP anymore, so I see if I can change it here.

Reply:
Actually, I remembered another reason why we did not do it before - `spawn_local` does not allow to get the result of spawned future back.

Reply:
@aqrln Updated to `spawn_local` with oneshot channel